Restored LMS Data Loaders

clintonb opened the pull request:
[ECOM-3982](https://openedx.atlassian.net/browse/ECOM-3982)

clintonb commented:
@bderusha @peter-fogg please review

peter-fogg commented on course_discovery/apps/course_metadata/data_loaders.py:
```diff
@@ -0,0 +1,175 @@
+        return s.strip() or None
+
+    @classmethod
+    def clean_strings(cls, data):
```
This should just be a static method, since it only needs `cls` to call a static method.

bderusha replied:
Did we come to any conclusion about this comment?

clintonb replied:
It's not changing. I'd rather use `cls` than `AbstractDataLoader`.

peter-fogg replied:
But my point is that using `cls` versus `AbstractDataLoader` is exactly the same, since the static methods can't be overridden by subclasses. If that's the intended behavior then they should be class methods as well.

clintonb replied:
All `classmethod`s now.

peter-fogg replied:
![nice](http://www.reactiongifs.com/r/yok.gif)

peter-fogg commented:
Minor comment, otherwise :+1:

bderusha commented:
:+1:

clintonb commented:
@peter-fogg @bderusha please re-review. I updated the a foreign key, and updated the Course API loader to retrieve pacing, image, and video information.

peter-fogg commented on course_discovery/apps/course_metadata/data_loaders.py:
```diff
@@ -0,0 +1,204 @@
+        # which may not be unique for an organization.
+        course_run_key = CourseKey.from_string(body['id'])
+        organization, __ = Organization.objects.get_or_create(key=course_run_key.org)
+        course_key = '{org}+{course}'.format(org=organization.key, course=course_run_key.course)
```
There's no better solution now, but this goes against the whole idea of opaque keys. It may be good to have a more principled way to construct keys in the future.

bderusha replied:
Can we be more explicit about which part of the `course` we are using to construct the key rather than relying on the `str` method (which someone may change without realizing that they are also changing how we are constructing keys)

clintonb replied:
I'm not sure what your'e asking. `str` method?

bderusha replied:
Ah nvm.  I misread.  Carry on!

peter-fogg commented:
:+1:

bderusha commented:
:+1:
